lib/ansible/modules/packaging/os/yum_repository.py

```diff
@@ -424,10 +426,11 @@
     mirrorlist: http://mirrorlist.repoforge.org/el7/mirrors-rpmforge
     enabled: no
 
-- name: Remove repository
+- name: Remove repository (and clean up left-over metadata)
   yum_repository:
     name: epel
     state: absent
+  notify: yum-clean-all
```

Comment:
The reason I gave no example of the notification handler, is that I have the create the complete playbook and all other examples were tasks.